src/catalog/catalog_accessor.cpp

```diff
@@ -217,4 +220,8 @@ common::ManagedPointer<storage::BlockStore> CatalogAccessor::GetBlockStore() con
   return catalog_->GetBlockStore();
 }
 
+void CatalogAccessor::RegisterTempTable(table_oid_t table_oid, const common::ManagedPointer<storage::SqlTable> table) {
+  temp_tables_[table_oid] = table;
```

Comment:
Similarly, there should be an assertion here that the oid is not already in the map, if that assumption is being made. Assert any assumptions about behavior so that when the next person goes to violate that assumption, they get an error.